airflow/utils/orm_event_handlers.py

```diff
@@ -86,3 +86,7 @@ def after_cursor_execute(conn, cursor, statement, parameters, context, executema
                 stack_info,
                 statement.replace("\n", " "),
             )
+
+    if conf.getboolean('core', 'execute_listeners_on_scheduler', fallback=False):
+        # On import, register sqlalchemy event handlers which call airflow.listeners
+        import airflow.listeners.events  # noqa
```

Comment:
IMO this is generally bad practice. I’d much prefer the import to not have side effect, and an explicit init function call registers the handlers. This would make tests much easier to write as well since you can do the init in a module-scoped pytest fixture instead.

Reply:
I definitely agree with @uranusjr . We already have some of those in Airflow and side effects have bitten us more than once.

Reply:
Fixed. Now it explicitely calls `register_task_instance_state_events`.